spring-cloud/spring-cloud-dataflow-acceptance-tests
# Isolate app's route paths between parallel running tests

Resolves #162

* Refactor AbstractStreamTests to:
  - Inline the debug messages from AbstractStreamTests#testResourceHandler into the LogTestNameRule
  - Add ExternalResources rule to handle the reliable stream destruction after test.
  - remove obsoleted fields from AbstractStreamTests

* Add random suffix to the SCDF Server name
  - In CF, in Classic mode the stream app-names have the following prefixes: `<SCDF SERVER APP NAME>-<STREAM NAME>`. Because in Classic mode there are no configuration properties to allow us set the APP_PREFIX_NAMES we should alter the SCDF Server's name in order to change the App Name's prefixes. That requires to add random suffix to the SCDF name in the CF manifest.
  - Because the SCDF server name has a dynamic part, make sure that the current name is retrieved from CF rather than using static 'scdf-server' label. Affected scripts: cloudfoundry/server/{config.sh, destroy.sh}
  - The Configuration Server uses the the <app-name>/<profile> convention to locate the application's properties. Because the SCDF server name has  dynamic suffix we have to use the `SPRING_CLOUD_CONFIG_NAME: scdf-server` property to override the default lookup convention. This will point to the existing `scdf-server-cloud1.properties` file (assuming prifle=cloud1).
  - Set the `app.log.spring.cloud.config.name=scdf-server-TICKTOCK-config-server-log` for the TickTock#tickTockTestsFromConfigServer tests to compensate the dynamic SCDF server name

* Skipper allows to set the App Name prefix via the `SPRING_CLOUD_SKIPPER_SERVER_PLATFORM_CLOUDFOUNDRY_ACCOUNTS[pws]_DEPLOYMENT_APP_NAME_PREFIX` property. To ensure isolations between CF Spaces we can set the app prefix to: $SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SPACE

* Update the CF env.properties with PWS as default values
* Improve documentation and add shell debug messages
* Use the `cf cf delete-orphaned-routes -f` to remove the orphaned paths after the test run
